server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -447,10 +450,23 @@ public SegmentIdentifier inTransaction(Handle handle, TransactionStatus transact
             // Avoiding ON DUPLICATE KEY since it's not portable.
             // Avoiding try/catch since it may cause inadvertent transaction-splitting.
 
+            // UNIQUE key for the row, ensuring sequences do not fork in two directions.
+            // Using a single column instead of (sequence_name, sequence_prev_id) as some MySQL storage engines
+            // have difficulty with large unique keys (see https://github.com/druid-io/druid/issues/2319)
+            final String sequenceNamePrevIdSha1 = BaseEncoding.base16().encode(
+                Hashing.sha1()
+                       .newHasher()
+                       .putBytes(sequenceName.getBytes(Charsets.UTF_8))
+                       .putByte((byte) 0xff)
+                       .putBytes(previousSegmentIdNotNull.getBytes(Charsets.UTF_8))
```

Comment:
ah okay, I was looking for that but hadn't found it. I only found SerializerUtils which wasn't statics and didn't seem right.